docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+
+.. code-block:: yaml
+
+    default:
+        suites:
+            ui_managing_countries:
+                contexts_as_services:
+                    - sylius.behat.context.hook.doctrine_orm //This service is responsible for clearing database before scenario.
+
+                    - sylius.behat.context.transform.country
+                    - sylius.behat.context.transform.shared_storage
+                    // Those contexts services are responsible for all transformation for example  "And the country "France" should appear in the store" transforms "(the country "France")" to proper Country object.
+
+                    - sylius.behat.context.setup.geographical
+                    - sylius.behat.context.setup.security
+                    // Setup contexts in our case they are building our background scene. Preparing all actors.
+
+                    // Lights, Camera, Action!
+                    - sylius.behat.context.ui.admin.managing_countries
+                    - sylius.behat.context.ui.admin.notification
+                    // Those contexts are essential here we are placing all action steps like "When I choose "France" and I add it Then I should ne notified that...".
+                filters:
+                    tags: "@managing_countries && @ui"
+
+First essential thing that happens here is configuration of tags, from now behat will be searching all your features tagged with ``@managing_countires`` and your scenarios tagged with ``@ui``.
+Second thing is ``contexts_as_services:`` in this section we will be placing all our services with step implementation.
+
+I mentioned about side effects of ubiquitous language that's why we can very easy switch our testing context to ``@domain``. Domain suite will look like:
+
+.. code-block:: yaml
+
+    default:
+        suites:
+            domain_managing_countries:
+                contexts_as_services:
+                    - sylius.behat.context.hook.doctrine_orm
+
+                    - sylius.behat.context.transform.country
+                    - sylius.behat.context.transform.shared_storage
+
+                    - sylius.behat.context.setup.geographical
+                    - sylius.behat.context.setup.security
+
+                    - sylius.behat.context.domain.admin.managing_countries // Domain step implementation.
+                filters:
+                    tags: "@managing_countries && @domain"
+
+We have almost finished with suite configuration now we need to register our first behat context service, but before we will do it we need one more thing because like we said this is ui feature so we need page objects.
+
+Registering Pages
+-----------------
+Page object pattern allows us to hide all interaction with html behind those objects so all ui things happens here.
+
+We got three kinds of page:
+    - Page // First layer of our pages it knows how to interact with dom objects. It has method ->getUrl(array $urlParameters) where you can define raw url to open it.
```

Comment:
- Page // First layer of our pages it knows how to interact with DOM objects. It has a method ->getUrl(array $urlParameters) where you can define a raw url to open it.